src/main/java/de/tum/in/www1/exerciseapp/security/JiraAuthenticationProvider.java

```diff
@@ -114,6 +114,7 @@ public User getOrCreateUser(Authentication authentication, Boolean skipPasswordC
 
             Optional<User> matchingUser = userService.getUserWithAuthoritiesByLogin(username);
             if (matchingUser.isPresent()) {
+                courseService.initTutorial(matchingUser.get());
```

Comment:
Since we already have users registered I think to check at login if the tutorials participation already exist is the right way to go